unit_test.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+TEST_RESULT=0
+DOCKERFILE='Dockerfile-test'
+IMAGE='tracker'
+POSTGRES_IMAGE='quay.io/cloudservices/postgresql-rds:15-1'
+TEARDOWN_RAN=0
+
+
+teardown() {
+
+    [ "$TEARDOWN_RAN" -ne "0" ] && return
+
+    echo "Running teardown..."
+
+    docker rm -f "$TEST_CONTAINER_NAME"
+
+    # remove postgres container
+    docker rm -f postgres
+    TEARDOWN_RAN=1
+}
+
+trap teardown EXIT ERR SIGINT SIGTERM
+
+mkdir -p artifacts
+
+get_N_chars_commit_hash() {
+
+    local CHARS=${1:-7}
+
+    git rev-parse --short="$CHARS" HEAD
+}
+
+TEST_CONTAINER_NAME="tracker-$(get_N_chars_commit_hash 7)"
+
+echo -e "\n---------------------------------------------------------------\n"
+
+echo "Pulling postgres image"
+docker pull $POSTGRES_IMAGE
+
+echo -e "\n---------------------------------------------------------------\n"
+
+echo "Starting postgres container"
+docker run --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5 -d --rm --name postgres -e POSTGRES_PASSWORD=crc -e POSTGRES_USER=crc -e POSTGRES_DB=crc -h 0.0.0.0:5432:5432 $POSTGRES_IMAGE
+sleep 5
+
+echo -e "\n---------------------------------------------------------------\n"
+
+echo "Building image"
+docker build -f "$DOCKERFILE" -t "$TEST_CONTAINER_NAME" .
+
+echo -e "\n---------------------------------------------------------------\n"
+
+echo "Running container"
+docker run -d --rm --name "$TEST_CONTAINER_NAME" "$IMAGE" sleep infinity
+
+echo -e "\n---------------------------------------------------------------\n"
+
+echo "Installing dependencies"
+docker exec --workdir /workdir "$TEST_CONTAINER_NAME" make install > 'artifacts/install_logs.txt'
```

Comment:
I don't think this is necessary.  I think the `build` step will install the dependencies.